docs/proposals/scheduledjob.md

```diff
@@ -0,0 +1,161 @@
+and time part according to ISO8601 format.
+
+The new `ScheduledJob` object will have the following content:
+
+```go
+// ScheduledJob represents the configuration of a single scheduled job.
+type ScheduledJob struct {
+    TypeMeta
+    ObjectMeta
+
+    // Spec is a structure defining the expected behavior of a job, including the schedule.
+    Spec ScheduledJobSpec
+
+    // Status is a structure describing current status of a job.
+    Status ScheduledJobStatus
+}
+
+// ScheduledJobList is a collection of scheduled jobs.
+type ScheduledJobList struct {
+    TypeMeta
+    ListMeta
+
+    Items []ScheduledJob
+}
+```
+
+`ScheduledJobSpec` structure is defined to contain all the information how the actual
+job execution will look like, including the `JobSpec` from [Job Controller API](job.md)
+and the schedule in ISO8601 format.  This implies that each ScheduledJob execution
+will be created from the JobSpec actual at a point in time when the execution will
+be started.
+
+```go
+// ScheduledJobSpec describes how the job execution will look like and when it will actually run.
+type ScheduledJobSpec struct {
+
+    // JobSpec is a structure defining the expected behavior of a job.
+    JobSpec JobSpec
+
+    // Schedule contains the schedule in ISO8601 format, eg.
+    // - 2015-07-21T14:00:00Z - represents date and time in UTC
+    // - R/2015-07-21T14:00:00Z/P1D - represents endlessly repeating interval (1 day), starting from given date
+    Schedule string
+
+    // ConcurrentPolicy specifies how to treat concurrent executions of a Job.
+    ConcurrentPolicy ConcurrentPolicy
+
+    // BlockOnFailure suspends scheduling of next job runs after a failed one.
+    BlockOnFailure bool
+}
+
+// ConcurrentPolicy describes how the job will be handled.
+// Only one of the following concurrent policies may be specified.
+// If none of the following policies is specified, the default one
+// is AllowConcurrent.
+type ConcurrentPolicy string
+
+const (
+    // AllowConcurrent allows ScheduledJobs to run concurrently.
+    AllowConcurrent ConcurrentPolicy = "AllowConcurrent"
+
+    // SkipNextConcurrent forbids concurrent runs, skipping next run if previous
+    // hasn't finished yet.
+    SkipNextConcurrent ConcurrentPolicy = "SkipNextConcurrent"
+
+    // CancelPreviousConcurrent forbids concurrent runs, canceling currently
+    // running job and replacing it with a new one.
+    CancelPreviousConcurrent ConcurrentPolicy = "CancelPreviousConcurrent"
+)
+```
+
+`ScheduledJobStatus` structure is defined to contain information about scheduled
+job executions.  The structure holds a list of currently running job instances
+and additional information about overall successful and unsuccessful job executions.
+
+```go
+// ScheduledJobStatus represents the current state of a Job.
+type ScheduledJobStatus struct {
+
+    // Jobs are the currently running instances of a ScheduledJob.
+    Jobs []Job
```

Comment:
Yeah, I haven't thought about sorting them by CreationTimestamp, which generally speaking allows me to distinguish (that's the word I've meant to use, sorry :/) different ScheduledJob executions/runs needed to properly handle ConcurrentPolicy.